### PR TITLE
Implement a compat function to get OID of a constraint

### PR DESCRIPTION
**Pull request by onurctirtir**
We can find OID of a constraint object by directly reading the related field from the Form_pg_constraint struct if PostgreSQL version >= 12.0 with no additional catalog table scan.

However, for the older versions, we need to utilize `HeapTupleGetOid` API.

This pr introduces a compat function for doing this.

### CODE REVIEW

**Comment by serprex** [comment]
Why can't PG11 be similarly optimized with `HeapTupleGetOid(heapTuple)` where we get the form form?

**Comment by onurctirtir** [comment]
> Why can't PG11 be similarly optimized with HeapTupleGetOid(heapTuple) where we get the form form?

It even does make more sense, updating the code. Thanks Philip

File: src/backend/distributed/commands/foreign_constraint.c [hunk 1]
```diff
@@ -536,6 +535,31 @@ GetTableForeignConstraintCommands(Oid relationId)
 }
 
 
+/*
+ * get_relation_constraint_oid_compat returns OID of the constraint represented
```
Comment (serprex) [inline src/backend/distributed/commands/foreign_constraint.c:539]: Comment should be updated to reflect that we can still retrieve the OID directly in pg11

Comment (onurctirtir) [reply src/backend/distributed/commands/foreign_constraint.c:539]: addressed

**Pull request merged**